src/pages/Editor.vue

```diff
@@ -811,24 +829,42 @@ export default {
     },
     addNewItem() {
       this.$refs.playerObj.player.pause();
-      // get all the data needed to create a new item
       var newItem = {};
+
+      // get a valid timestamp for adding a new item
       var newTimestamp = this.getTimestampForNewItem();
       if (newTimestamp == -1) {
         this.showCannotAddItemDialog();
         return;
       }
-      // create the newItem object
-      newItem["time"] = newTimestamp;
-      newItem["type"] = this.getItemTypeForNewItem();
-      newItem["meta"] = this.getMetadataForNewItem();
-      newItem["details"] = this.getDetailsForNewItem();
 
-      // push it into items, update the itemTimestamps and currentItemIndex
-      this.items.push(newItem);
-      this.itemTimestamps = this.getItemTimestamps(this.items);
-      this.currentItemIndex = this.itemTimestamps.indexOf(this.currentTimestamp);
-      this.markItemSelected(this.currentItemIndex);
+      // create item, then create the question, then update local states
+      ItemService.createItem({
+        plio: this.plioDBId,
+        type: this.getItemTypeForNewItem(),
+        time: newTimestamp,
+        meta: this.getMetadataForNewItem(),
+      })
+        .then((createdItem) => {
+          newItem = createdItem;
+
+          if (createdItem.type == "question") {
+            return QuestionService.createQuestion({
+              item: createdItem.id,
+              type: "mcq",
+              options: ["", ""],
+              correct_answer: "0",
+            });
+          }
+        })
+        .then((createdQuestion) => {
+          newItem.details = createdQuestion;
+          // push it into items, update the itemTimestamps and currentItemIndex
+          this.items.push(newItem);
```

Comment:
this does not care about the ordering of the items based on timestamp

Reply:
Sorting is being taken care of here. Pushing into `this.items` does not sort inherently, but as it goes to the next line,
```
this.itemTimestamps = this.getItemTimestamps(this.items);
```
the `itemTimestamps` array is updated - still unsorted.

There is a `watch:` method on `itemTimestamps` - 
```
itemTimestamps() {
      this.itemTimestamps.forEach((itemTimestamp, index) => {
        this.items[index]["time"] = itemTimestamp;
      });
      // handle item sorting and marker positioning
      // when time is changed from the time input boxes
      // or when item is added using the add item button
      this.handleTimeUpdateFromEditor();
    },
```
Whenever `itemTimestamps` change, the sorting is handled by the function `handleTimeUpdateFromEditor()`

Although I agree the name of the function can be changed. This function in general sorts the items according to the new timestamps and resets the `currentItemIndex`. Please suggest a better name

Reply:
Got it. My bad. 
We can use `checkAndFixItemOrder`